test/Data/Source/Spec.hs

```diff
@@ -68,6 +72,10 @@ testTree = Tasty.testGroup "Data.Source"
     ]
 
   ]
+  where summarize src = case sourceLines src of
+          []  -> "empty"
+          [x] -> if nullSource x then "empty" else "single-line"
+          _   -> "multiple lines"
```

Comment:
We could do some more summarizing w.r.t. the specific line endings in use, but as we actually use `sourceLineRanges` almost exclusively in error reporting from `Assignment`, I didn’t figure it was worth the effort.